cpp/src/barretenberg/proof_system/composer/permutation_lib.hpp

```diff
@@ -96,6 +94,32 @@ std::vector<CyclicPermutation> compute_wire_copy_cycles(const typename Flavor::C
         }
     }
 
+    // Define offsets for placement of public inputs and gates in execution trace
+    const size_t num_zero_rows = Flavor::has_zero_row ? 1 : 0;
+    size_t pub_inputs_offset = num_zero_rows;
+    size_t gates_offset = num_public_inputs + num_zero_rows;
+
+    // If Goblin, adjust offsets to account for ecc op gates and update copy cycles to include these gates
+    if constexpr (IsGoblinFlavor<Flavor>) {
+        // Set ecc op gate offset and update offsets for PI and conventional gates
+        const size_t op_gates_offset = num_zero_rows;
+        const size_t num_ecc_op_gates = circuit_constructor.num_ecc_op_gates;
+        pub_inputs_offset += num_ecc_op_gates;
+        gates_offset += num_ecc_op_gates;
+
+        const auto& op_wires = circuit_constructor.ecc_op_wires;
+        // Iterate over all variables of the ecc op gates, and add a corresponding node to the cycle for that variable
+        for (size_t i = 0; i < num_ecc_op_gates; ++i) {
+            // Note: We exclude the first op wire since it contains the op codes which are not stored in variables
```

Comment:
Are you sure about that? You'll have to fix the operation in some way or other. One of the ways to do that is to create a cycle, connecting all the same operations together and to a known constant. Without that mechanism you'll have to fix them with some other relation.

Reply:
Good point. I'm not sure what the best approach is. I like the idea of using a copy constraint to constrain to constant values. As you say, it should be easy enough to retrofit a solution. I added a TODO that reflects your suggestions.